src/application/notifications.ts

```diff
@@ -0,0 +1,56 @@
+import { User } from "../domain/user.js";
+import { Authentication, EmailClient } from "../infrastructure/email.js";
+import { SecretsClient } from "../infrastructure/secrets.js";
+
+export class NotificationsService {
+  private readonly sender = process.env.NOTIFICATIONS_EMAIL;
+  constructor(
+    private readonly emailClient: EmailClient,
+    private readonly secretsClient: SecretsClient
+  ) {}
+
+  private async populateAuth() {
+    if (this.emailClient.auth) {
+      return;
+    }
+
+    const [clientId, clientSecret, refreshToken] = await Promise.all([
+      this.secretsClient.accessSecret("notifications-email-client-id"),
+      this.secretsClient.accessSecret("notifications-email-client-secret"),
+      this.secretsClient.accessSecret("notifications-email-refresh-token"),
+    ]);
+
+    const auth: Authentication = {
+      type: "OAuth2",
+      accessUrl: "https://accounts.google.com/o/oauth2/token",
+      user: this.sender,
+      clientId,
+      clientSecret,
+      refreshToken,
+    };
+
+    this.emailClient.auth = auth;
+  }
+
+  public async notifySuccess(recipient: User): Promise<void> {
+    await this.populateAuth();
+    await this.emailClient.sendEmail(
+      "smtp.gmail.com",
+      recipient.email,
+      this.sender,
+      "Publish to BCR [no-reply]",
+      "Successfully published pull request to the Bazel Central Registry."
```

Comment:
Link to the PR?

Reply:
This is just placeholder comment. I'm planning to do a pass on the email content in another PR.